### PR TITLE
Upgrade node to v10.8.0 for improved memory management

### PR DESCRIPTION
**Pull request by phillipperalez**
Local testing has seen a large improvement in memory management. 

Upgrades Node to v10.8.0. v10 will be LTS around October

### CODE REVIEW

**Comment by wilg** [comment]
You will need to change `.nvmrc` also

**Comment by wilg** [comment]
And the `Dockerfile`

File: src/actions/segment/segment.ts [hunk 1]
```diff
@@ -215,7 +215,7 @@ export class SegmentAction extends Hub.Action {
   }
 
   protected segmentClientFromRequest(request: Hub.ActionRequest) {
-    return new segment(request.params.segment_write_key)
+    return new segment(request.params.segment_write_key, {flushAt: 500})
```
Comment (wilg) [inline src/actions/segment/segment.ts]: Can we put this in a separate PR for clarity?

Comment (wilg) [reply src/actions/segment/segment.ts]: Or at least its own commit

Comment (phillipperalez) [reply src/actions/segment/segment.ts]: for sure I'll take that out for this

Comment (wilg) [reply src/actions/segment/segment.ts]: or rebase the whole thing so that the node upgrade and the segment flush are separate things.

and i'd still like to make this an environment variable with a default 500

Comment (wilg) [reply src/actions/segment/segment.ts]: ok cool yea a seperate PR is best that way this one can be squashed

**Comment by wilg** [comment]
The test failure is strange – maybe the Node upgrade changes something subtle?
